server/base/src/main/java/org/apache/accumulo/server/manager/state/TabletMetadataIterator.java

```diff
@@ -0,0 +1,450 @@
+      case SUSPENDED:
+      case UNASSIGNED:
+        if (shouldBeOnline && (tm.getHostingGoal() == TabletHostingGoal.ALWAYS
+            || (tm.getHostingGoal() == TabletHostingGoal.ONDEMAND && tm.getHostingRequested()))) {
+          return true;
+        }
+        break;
+      default:
+        throw new AssertionError(
+            "Inconceivable! The tablet is an unrecognized state: " + getLocationState(current, tm));
+    }
+    return false;
+  }
+
+  private static TabletState getLocationState(final Set<TServerInstance> liveServers,
+      final TabletMetadata tm) {
+    Location loc = tm.getLocation();
+
+    if (loc == null || loc.getType() == null || loc.getServerInstance() == null) {
+      return TabletState.UNASSIGNED;
+    }
+
+    if (loc.getType().equals(LocationType.FUTURE)) {
+      return liveServers.contains(loc.getServerInstance()) ? TabletState.ASSIGNED
+          : TabletState.ASSIGNED_TO_DEAD_SERVER;
+    } else if (loc.getType().equals(LocationType.CURRENT)) {
+      return liveServers.contains(loc.getServerInstance()) ? TabletState.HOSTED
+          : TabletState.ASSIGNED_TO_DEAD_SERVER;
+    } else if (tm.getSuspend() != null) {
+      return TabletState.SUSPENDED;
+    } else {
+      return TabletState.UNASSIGNED;
+    }
+  }
+
+  public static void configureScanner(final ScannerBase scanner, final CurrentState state) {
+    TabletColumnFamily.PREV_ROW_COLUMN.fetch(scanner);
+    scanner.fetchColumnFamily(CurrentLocationColumnFamily.NAME);
+    scanner.fetchColumnFamily(FutureLocationColumnFamily.NAME);
+    scanner.fetchColumnFamily(LastLocationColumnFamily.NAME);
+    scanner.fetchColumnFamily(SuspendLocationColumn.SUSPEND_COLUMN.getColumnFamily());
+    scanner.fetchColumnFamily(LogColumnFamily.NAME);
+    scanner.fetchColumnFamily(ChoppedColumnFamily.NAME);
+    scanner.fetchColumnFamily(HostingColumnFamily.NAME);
+    scanner.addScanIterator(new IteratorSetting(1000, "wholeRows", WholeRowIterator.class));
+    IteratorSetting tabletChange =
+        new IteratorSetting(1001, "tabletMetadataIterator", TabletMetadataIterator.class);
+    if (state != null) {
+      TabletMetadataIterator.setCurrentServers(tabletChange, state.onlineTabletServers());
+      TabletMetadataIterator.setOnlineTables(tabletChange, state.onlineTables());
+      TabletMetadataIterator.setMerges(tabletChange, state.merges());
+      TabletMetadataIterator.setMigrations(tabletChange, state.migrationsSnapshot());
+      TabletMetadataIterator.setManagerState(tabletChange, state.getManagerState());
+      TabletMetadataIterator.setShuttingDown(tabletChange, state.shutdownServers());
+    }
+    scanner.addScanIterator(tabletChange);
+  }
+
+  public static TabletMetadata decode(Entry<Key,Value> e) throws IOException {
+    final SortedMap<Key,Value> decodedRow = WholeRowIterator.decodeRow(e.getKey(), e.getValue());
+    return TabletMetadata.convertRow(decodedRow.entrySet().iterator(), CONFIGURED_COLUMNS, false,
+        true);
+  }
```

Comment:
The reason information is more associated with this iterator than it is with TabletMetadata.  Could move that code to this iterator.  Below was something I was playing around.  If this makes sense could possibly rename this iterator to TabletAlertIterator

```suggestion
  // was trying to think of a single word for "needs attention" and Alert was the best thing I could think of... 
  public static class TabletAlert {


    public static enum Reasons {
      BAD_STATE,
      NEEDS_COMPACTING,
      NEEDS_LOCATION_UPDATE,
      IS_MERGING,
      IS_MIGRATING,
      NEEDS_SPLITTING;
    }

    public static final String REASONS_COLUMN_NAME = "REASONS";

    public final Set<Reasons> reasons;
    public final TabletMetadata tabletMetadata;

    public TabletAlert(Set<Reasons> reasons, TabletMetadata tabletMetadata) {
      this.reasons = reasons;
      this.tabletMetadata = tabletMetadata;
    }

  }

  public static TabletAlert decode(Entry<Key,Value> e) throws IOException {
    final SortedMap<Key,Value> decodedRow = WholeRowIterator.decodeRow(e.getKey(), e.getValue());

    var row = decodedRow.firstKey().getRow();
    // must remove the alert reasons before creating the tablet metadata object
    var value = decodedRow.remove(new Key(row, TabletAlert.REASONS_COLUMN_NAME, ""));

    Set<TabletAlert.Reasons> reasons = decodeReasons(value);

    var tabletMetadata = TabletMetadata.convertRow(decodedRow.entrySet().iterator(), CONFIGURED_COLUMNS, false,
        true);

    return new TabletAlert(reasons, tabletMetadata);
  }
```

Reply:
You used the word Management in another type name and I like that. I was trying to think of how to work the word Management into this type name. Was thinking of things like `TabletManagementInfo`, `TabletManagementAction`, `TableManagementAlert`... not really sure about these, but those were things I thought of.